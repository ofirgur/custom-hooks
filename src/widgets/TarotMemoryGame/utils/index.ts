export const isMobileScreen = () => window.matchMedia('(max-width: 500px)').matches;
