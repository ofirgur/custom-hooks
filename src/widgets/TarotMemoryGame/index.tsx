import React, { useEffect } from 'react';
import { INITIAL_GAME_STATE, GAME_STATE_LOCAL_STORAGE_NAME } from './utils/constants';
import { getGameStateLocalStorage, removeGameStateLocalStorage } from '../../localstorage';
import GameOutletScreen from './components/GameOutletScreen';
import { GameStateContextProvider } from './utils/context';

const TarotMemoryGame = () => {
  const GAME_STATE_LOCAL_STORAGE = getGameStateLocalStorage(GAME_STATE_LOCAL_STORAGE_NAME);
  useEffect(() => {
    removeGameStateLocalStorage(GAME_STATE_LOCAL_STORAGE_NAME);
  }, []);

  return (
    <GameStateContextProvider initialState={GAME_STATE_LOCAL_STORAGE || INITIAL_GAME_STATE}>
      <GameOutletScreen />
    </GameStateContextProvider>
  );
};

export default TarotMemoryGame;
