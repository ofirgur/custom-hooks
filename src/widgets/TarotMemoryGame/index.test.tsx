import * as React from 'react';
import { render } from '@testing-library/react';

import TarotMemoryGame from './';

describe('TarotMemoryGame', () => {
  it('should render correctly', () => {
    const { container } = render(<TarotMemoryGame />);
    expect(container).toMatchSnapshot();
  });
});
