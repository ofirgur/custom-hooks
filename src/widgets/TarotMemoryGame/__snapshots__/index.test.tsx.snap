// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TarotMemoryGame should render correctly 1`] = `
<div>
  <div
    class="css-6l25fd"
  >
    <div
      class="css-1j3umj3"
    >
      <div>
        <div
          class="css-1wt90mm"
        >
          ADVERTISEMENT
        </div>
        <ins
          class="adsbygoogle css-1ujbgzu"
          data-ad-client="ca-pub-2910351605291781"
          data-ad-slot="8521368831"
        />
      </div>
      <h4
        class="css-ujyrfm"
      >
        Select A Level
      </h4>
      <div
        class="css-1gm5two"
      >
        <button
          class="css-1i01t95"
          role="button"
        >
          <img
            alt=""
            class="css-9rlldu"
            src="beginner.png"
          />
          <span
            class="css-1kp7frp"
          >
            Beginner
          </span>
        </button>
        <button
          class="css-1i01t95"
          role="button"
        >
          <img
            alt=""
            class="css-9rlldu"
            src="intermediate.png"
          />
          <span
            class="css-1kp7frp"
          >
            Intermediate
          </span>
        </button>
        <button
          class="css-1i01t95"
          role="button"
        >
          <img
            alt=""
            class="css-9rlldu"
            src="advanced.png"
          />
          <span
            class="css-1kp7frp"
          >
            Advanced
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
